### PR TITLE
[no gbp] Fixes runtimes in shuttle event and shuttle cling

### PR DESCRIPTION
**Pull request by Time-Green**
Fixes no-candidate runtime from pick() on an empty list
Fixes spawning null movables if run out of mobs to spawn
Fixes shuttle cling qdeling on Initialize due to lattices not counting as hyperspace

:cl:
runtime: Fixes runtimes in shuttle event and shuttle cling code
/:cl:

### CODE REVIEW

File: code/game/turfs/open/space/transit.dm [hunk 1]
```diff
@@ -36,13 +36,13 @@
 /turf/open/space/transit/proc/initialize_drifting(atom/entered, atom/movable/enterer)
 	SIGNAL_HANDLER
 
-	if(enterer && !HAS_TRAIT(enterer, TRAIT_HYPERSPACED))
+	if(enterer && !HAS_TRAIT(enterer, TRAIT_HYPERSPACED) && !(locate(/obj/structure/lattice) in contents))
```
Comment (Jacquerel) [inline code/game/turfs/open/space/transit.dm:39]: Can we make `lattice` apply a trait to the turf instead? I think we have an element or component for this.

**Review by Jacquerel — CHANGES_REQUESTED** [review]

File: code/game/turfs/open/space/transit.dm (hunk 1 above)
Comment (Time-Green) [reply code/game/turfs/open/space/transit.dm]: Yes! I changed the istypes to use that element with TRAIT_HYPERSPACE_STOPPED

File: code/game/objects/structures/lattice.dm [hunk 2]
```diff
@@ -154,7 +154,7 @@
 	canSmoothWith = SMOOTH_GROUP_LATTICE
 	obj_flags = CAN_BE_HIT | BLOCK_Z_OUT_DOWN | BLOCK_Z_IN_UP
 	resistance_flags = FIRE_PROOF | LAVA_PROOF
-	give_turf_traits = list(TRAIT_LAVA_STOPPED, TRAIT_CHASM_STOPPED, TRAIT_IMMERSE_STOPPED)
+	give_turf_traits = list(TRAIT_LAVA_STOPPED, TRAIT_CHASM_STOPPED, TRAIT_IMMERSE_STOPPED, TRAIT_CHASM_STOPPED)
```
Comment (Jacquerel) [inline code/game/objects/structures/lattice.dm:157]:
```suggestion
	give_turf_traits = list(TRAIT_LAVA_STOPPED, TRAIT_CHASM_STOPPED, TRAIT_IMMERSE_STOPPED, TRAIT_HYPERSPACE_STOPPED)
```
Copy/paste error here I think.

**Review by Jacquerel — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/lattice.dm (hunk 2 above)
Comment (Time-Green) [reply code/game/objects/structures/lattice.dm]: 🫠

**Pull request merged**